meetings/working-groups/extensions/extending-extensions-a-guide-to-relaxation.md

```diff
@@ -0,0 +1,316 @@
+By introducing the concept of an "extension declaration" declared anonymously within a static class to serve as a
+grouping for extension members, it elegantly avoids the "type explosion" problem that has plagued other designs. This
+new design ensures that extension members can be declared with a simple, familiar syntax while removing any rough edges
+that would complicate consumption.
+
+The design is based on a set of assumptions that are not necessarily agreed upon, but impose restrictions that bring
+clarity and simplicity to the design space. It is the belief of this author that some of these assumptions will not be
+true for all and the feature will feel restrictive in its current form.
+
+Generally, the restrictions will be felt by those declaring extension members, not those consuming extension members.
+Classic extension methods offer a tremendous amount of flexibility with regard to the declaration and organization of
+methods. In contrast, anonymous extension declarations design imposes limits that may serve as speed bumps to existing
+extension method authors. It is the goal of this proposal to identify opportunities to relax some of those restrictions
+with the goal of providing a similar level of flexibility for anonymous extension declarations that classic extension
+methods already enjoy.
+
+This is approach is not unlike C# records. Positional records provide a simple and straightforward syntax that offer
+type authors a broad set of functionality. However, during the design of records, it was realized that type authors
+needed ways to customize records in various ways. And so, many features were introduced for type authors to serve as
+"knobs" when declaring records, such as init-only properties, Deconstruct methods, value-based equality customization,
+required properties, mutable and immutable struct-based records, etc.
+
+## Detailed design
+
+### Optional parameter name on an extension declaration
+
+The anonymous extension declarations proposal makes the following assumption:
+
+> - Parameter names for underlying values aren't important and people will resent the forced verbosity of having to specify them.
+
+While this leads to a clear design, that assertion will not be true for all. Parameter names provide important value
+to a member’s signature and the code within.
+
+- Parameter names are an important form of self-documentation when the type system isn’t expressive enough to clearly
+  state the author's intent. For example, it is helpful when a parameter of type `Person` has the parameter name
+  `teacher`. This is a clear indicator to the caller what sort of `Person` to pass and helps make the code within the
+  member more readable.
+- Parameter names can synergize semantically with other parameter names. For example, the parameter name `source` might
+  be linked semantically to another parameter named `destination`.
+- Parameter names can be used to disambiguate overloads by applying a named argument.
```

Comment:
Though, this would not apply when used *as* an extension member, and would not apply at all unless it was decided that these things could be called as ordinary members.

Reply:
That's true! Here, I'm trying to list reasons why parameter names are generally important. Using a `this` parameter name as a named argument for disambiguation of classic extension method is definitely super-rare, but it is certainly possible.

On a side note, I think it's an open question to allow any new disambiguation form to work with classic extension methods or the classic static-method-call to work with new extension methods. However, I would advocate for making both cases work if only because consumers shouldn't have to know how the extension method was implemented. To the consumer, it should just be an extension method.